### PR TITLE
pwm: pwm_nrf5_sw: Add prescaler support and use it in the servo_motor sample

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
These commits fix the inaccuracies regarding PWM frequency and pulse width in the current pwm_nrf5_sw implementation, adds support for specifying a HF timer prescaler value and finally uses the prescaler to enable use of the servo_motor sample on the BBC micro:bit.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]

**Comment by dbkinder** [comment]
you'll need to fix the commit message title (too long)

**Comment by henrikbrixandersen** [comment]
@dbkinder Thank you for reviewing. Which commit has the too long title?

**Comment by henrikbrixandersen** [comment]
This is an alternative solution to #7474 which is more in line with the proposed addition of the nrfx PWM HW driver in #7334.

File: drivers/pwm/Kconfig.nrf5_sw [hunk 1]
```diff
@@ -22,3 +22,24 @@ config PWM_NRF5_SW_0_DEV_NAME
 	help
 	  Specify the device name for the Nordic Semiconductor nRF5x series S/W
 	  PWM driver.
+
+config PWM_NRF5_SW_0_CLOCK_PRESCALER
+	int
+	prompt "Nordic Semiconductor nRF5x series S/W PWM Clock Prescaler"
+	depends on PWM_NRF5_SW
```
Comment (cvinayak) [inline drivers/pwm/Kconfig.nrf5_sw]: use IF clause enclosing the DEV_NAME together, instead of using depends on (as the two can be grouped together now)

**Review by cvinayak — CHANGES_REQUESTED** [review]
Minor change requested, else LGTM.

**Comment by henrikbrixandersen** [comment]
@cvinayak Thank, fixed.

**Comment by findlayfeng** [comment]
我的本意是修复 period_cycles  大于 0xffff 时会引发的bug，在这里，我认为应当在period_cycles大于0xffff 时返回错误
From Google Translate
My intention is to fix the bug that is caused when period_cycles is greater than 0xffff. Here, I think it should return an error when period_cycles is greater than 0xffff.

**Comment by henrikbrixandersen** [comment]
@findlayfeng Good point, that should be addressed as well. Not sure I understand your math in #7474 regarding this, but perhaps it can just be rebased to address the overflow issue once this has been merged?

**Review by cvinayak — APPROVED** [review]

**Comment by cvinayak** [comment]
@carlescufi please do the needful.

**Comment by cvinayak** [comment]
@jhedberg hope you can check the mesh demo buzzer has no regression.

**Comment by jhedberg** [comment]
> @jhedberg hope you can check the mesh demo buzzer has no regression.

@cvinayak I tested with `samples/boards/microbit/sound/` - seems to work fine.

**Review by jhedberg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**